### DISCUSSION TITLE
Lightweight SIP / softphone client to enable full 2-ways audio on demand ?

### DISCUSSION THREAD

**Discussion by SebLz** [roleoroleo/yi-hack-MStar#440] [Q&A]
Hello,
I was wondering if anyone managed to get some lightweight SIP/softphone client to work on the mstar (or allwinner) cameras.
Since recently, there TECH7FOX managed to get 2-ways audio from HA working with a relatively simple setup, cf: https://tech7fox.github.io/sip-hass-docs/.
It would be great if we could use the cameras as kind of interphones/doorbells and the only missing part is a lightweight, command-line  SIP client that would run on the camera platform.

**Comment by SebLz** [comment]
I guess Linephone is what I'm looking for, but I'm struggling to cross-compile it for the mstar/allwinner architecture.
All instructions from the official wiki for raspi should be ok otherwise:
https://wiki.linphone.org/xwiki/wiki/public/view/Linphone/Linphone%20and%20Raspberry%20Pi/

**Comment by roleoroleo** [comment]
I also found other clients:
https://github.com/guisousanunes/sipcmd2
But there are 2 problems:
- build the required library.
- hack the source code to change the access to the speaker and the microphone.

It's a very hard job.

**Reply by SebLz** [reply]
Thanks for your reply and time @roleoroleo 
IMO this feature would bring a lot of value to the cameras but as you mentioned it, it might unfortunately be too complicated/time consuming to implement.
